### PR TITLE
serialize write operation

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by ngaut** [comment]
How about 3M rows?

File: tikv/region.go [hunk 1]
```diff
@@ -109,6 +119,32 @@ func (ri *regionInfo) marshal() []byte {
 	return data
 }
 
+func (ri *regionCtx) lock(hashVals []uint64) (bool, *sync.WaitGroup) {
```
Comment (ngaut) [inline tikv/region.go]: s/lock/aquireLocks

File: tikv/region.go [hunk 2]
```diff
@@ -109,6 +119,32 @@ func (ri *regionInfo) marshal() []byte {
...
+	}
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	for _, hashVal := range hashVals {
+		ri.memLocks[hashVal] = wg
+	}
+	return true, nil
+}
+
+func (ri *regionCtx) unlock(hashVals []uint64) {
```
Comment (ngaut) [inline tikv/region.go]: s/unlock/releaseLocks/

**Comment by coocood** [comment]
@ngaut PTAL

**Pull request merged**